### PR TITLE
API contract for Update Archived route

### PR DESCRIPTION
**Pull request by manish591**
## API contract for Update Archived route

Updates the user's `archived` property to true if the `in_discord` property is false.

### Current Users snapshot

```
  const users = [
    {
      first_name: 'name_one',
      last_name: 'last_one',
      roles: {
        archived: false,
        in_discord: false,
     }
    },
    {
      first_name: 'name_one',
      last_name: 'last_one',
      roles: {
        archived: false,
        in_discord: false,
     }
    },
  ]
```

### After updating

```
  const users = [
    {
      first_name: 'name_one',
      last_name: 'last_one',
      roles: {
        archived: true,
        in_discord: false,
     }
    },
    {
      first_name: 'name_one',
      last_name: 'last_one',
      roles: {
        archived: false,
        in_discord: false,
     }
    },
  ]
```

Added the API contract for the 2nd part of the task https://github.com/Real-Dev-Squad/website-members/issues/541

### CODE REVIEW

**Review by heyrandhir — COMMENTED** [review]
LGTM But can you check with @RitikJaiswal75 and @prakashchoudhary07 regarding the name of the API as in `/users/update-archived` only users is a resource `update-archived` is not a resource. Is there a better way we can name this API. I have seen action based API names in our various APIs itself but not sure if its right or not. 

@RitikJaiswal75 @prakashchoudhary07 @ankurnarkhede please suggest

File: users/README.md [hunk 1]
```diff
@@ -34,16 +34,17 @@ number and email address.
...
+| :---------------------------------------------------------: | :----------------------------------: |
+|                [GET /users](#get-users)                     |   Returns all users in the system    |
+|           [GET /users/self](#get-usersSelf)                 | Returns the logged in user's details |
+|  [GET /users/userId/:userId](#get-usersuseriduserid)        |    Returns user with given userId    |
+|       [GET /users/:username](#get-usersusername)            |   Returns user with given username   |
+|    [GET /users/:userId/badges](#get-usersidbadges)          | Returns badges assigned to the user  |
+|               [POST /users](#post-users)                    |          Creates a new User          |
+|         [PATCH /users/self](#patch-usersself)               |       Updates data of the User       |
+| [PATCH /users/:id/temporary/data](#patch-usersidroles)      |          Updates user roles          |
+| [PATCH /users/update-archived](#patch-usersupdate-archived) |   Archive users if not in discord    |
```
Comment (RitikJaiswal75) [inline users/README.md:47]: is this feature already merged? an update is an action I don't think should be a part of the API name.

File: users/README.md [hunk 2]
```diff
@@ -303,3 +304,28 @@ Updates roles for the User.
...
+- **Success Response:**
+  - **Code:** 200
+    - **Content:** `{ 'message': 'Successfully updated users archived role to true if in_discord role is false'}`
+- **Error Response:**
+  - **Code:** 401
+    - **Content:**
+      `{ 'statusCode': 401, 'error': 'Unauthorized', 'message': 'Unauthenticated User' }`
+  - **Code:** 500
+    - **Content:**
+      `{ 'statusCode': 500, 'error': 'Internal Server Error', 'message': 'An internal server error occurred' }`
```
Comment (RitikJaiswal75) [inline users/README.md:331]: please use multi line statements in code blocks mentioning proper language
it gives a cleaner look to the document.

Also add some examples.

File: users/README.md [hunk 3]
```diff
@@ -303,3 +304,28 @@ Updates roles for the User.
...
+Archive users if not in Discord.
+
+- **Params**  
+  None
+- **Query**  
+  None
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  rds-session: `<JWT>`
```
Comment (RitikJaiswal75) [inline users/README.md:319]: i think we should mention separate cookies for separate environments

File: users/README.md [hunk 4]
```diff
@@ -303,3 +304,28 @@ Updates roles for the User.
   - **Code:** 500
     - **Content:**
       `{ 'statusCode': 500, 'error': 'Internal Server Error', 'message': 'An internal server error occurred' }`
+
+## **PATCH /users/update-archived**
```
Comment (RitikJaiswal75) [inline users/README.md:308]: A heading level 2 is already bold any specific need to make it bold(er)

File: users/README.md (hunk 1 above)
Comment (manish591) [reply users/README.md:47]: No, It's not merged yet. Can you suggest a name?

Comment (apurvkhare) [reply users/README.md:47]: `/users/archived` should be fine

File: users/README.md [hunk 5]
```diff
@@ -303,3 +304,56 @@ Updates roles for the User.
...
+  None
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**
+  - **Prod**  
+    rds-session: `<SUPERUSER JWT>`
+  - **Development**  
+    rds-session-development: `<SUPERUSER JWT>`
+  - **Staging**  
+    rds-session-staging: `<SUPERUSER JWT>`
```
Comment (prakashchoudhary07) [inline users/README.md:324]: I think mention rds-session cookie would be suffice

File: users/README.md [hunk 6]
```diff
@@ -303,3 +304,88 @@ Updates roles for the User.
...
+  - Optional: `debug=[boolean]`(`debug` when set to true returns additional result in response to help debugging)
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  rds-session: `<SUPERUSER JWT>`
+- **Body**
+
+  ```json
+  {
+    // Defines action associated with PATCH /users route. Calls specific api based on action passed. If wrong payload is passed it throws an error
```
Comment (heyrandhir) [inline users/README.md:324]: inside a JSON we cant add comments hence this is highlighted in red. please move it above from where json starts

**Review by heyrandhir — CHANGES_REQUESTED** [review]
Please fix the formatting. the red highlight indicates the md file is not properly formatted.

**Review by heyrandhir — APPROVED** [review]

**Review by sahsisunny — APPROVED** [review]

**Pull request merged**